### PR TITLE
Support most remaining length/percentage/none/auto-taking properties in geckolib

### PR DESCRIPTION
**Pull request by heycam**
This is on top of #10999 so no need to review 8ae820b.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11001)

### CODE REVIEW

File: ports/geckolib/properties.mako.rs [hunk 1]
```diff
@@ -354,14 +358,20 @@ impl Debug for ${style_struct.gecko_ffi_name} {
    # These are booleans.
    force_stub += ["page-break-after", "page-break-before"]
 
-   simple_types = ["Number", "Opacity"]
+   # Properties that would be auto-generated but need to be manually implemented.
+   manual = ["bottom", "left", "right", "top"]
```
Comment (bholley) [inline ports/geckolib/properties.mako.rs]: I don't understand why we need this - doesn't this duplicate the purpose of the skip_longhands annotation that we have on the style struct declarations?

**Comment by bholley** [comment]
r=me with that fixed.

@bors-servo delegate+

File: ports/geckolib/properties.mako.rs (hunk 1 above)
Comment (heycam) [inline ports/geckolib/properties.mako.rs]: Ah of course, we can just list them there.

**Comment by heycam** [comment]
r=@bholley

**Pull request merged**